astronomer/providers/sftp/hooks/sftp.py

```diff
@@ -34,7 +36,7 @@ class SFTPHookAsync(BaseHook):
     default_conn_name = "sftp_default"
     conn_type = "sftp"
     hook_name = "SFTP"
-    default_known_hosts = "none"
+    default_known_hosts = "~/.ssh/known_hosts"
```

Comment:
Does this cause any problems if this file doesn't exist?

Reply:
yes, it expects this file to be present on the server. And complains that no known_hosts file found when trying to operate. We're mandating `host_key` check now by default as opposed to the sync version which by allows to proceed if no host key is given by just logging a warning for MITM attacks.

If a host key is not provided in the conn extras, we check for the host key in this file. If the needed host key is not provided in the connection extras or present in this file, we raise an exception. If the user wants to bypass this check, they can set `{no_host_key_check: true}` in the connection extras and in that case if this file is not present, it does not cause an issue.

Reply:
If I'm reading the code right (and I may not be) then we will call `_validate_host_key_using_paramiko` and that will always call `load_host_keys` -- this makes this a breaking change as it requires this file to exist even when host key was specified in the extras.

Reply:
This file must not be required if a hostkey is provided in extra - this makes it a DAG breaking change.

Reply:
As far as I could check, the asyncssh client that we use in the triggerer can only use host keys given in the known_hosts file unlike the paramiko client instance with which we can add a key to the client on top of the known_hosts file.

So, the approach I am taking is to validate the extras host_key using paramiko and using paramiko add the host key back to the known_hosts file so that the asyncssh client can perform its operations.

Do you have a suggestion for me on how we could tackle this scenario?

Reply:
hi @ashb , I believe I have addressed the concerns and verified that it does not break anymore asking for the `known_hosts` file to be present when connection `extras` `host_key` is provided in the connection. We have users waiting for these issues to be resolved that are being addressed in this PR and also have an open Zendesk ticket for a long time. So, as discussed in the providers standup today, we're thinking to go ahead with this and create a release. 
If you still foresee some issues, we can have a follow up PR. 

cc: @phanikumv